packages/astro/src/core/dev/index.ts

```diff
@@ -329,14 +328,7 @@ export class AstroDevServer {
       res.end();
     } catch (err: any) {
       const statusCode = 500;
-      const mod = filePath && this.viteServer.moduleGraph.getModuleById(viteifyURL(filePath));
-      if (mod) {
-        for (const m of [mod, ...mod.importedModules]) {
-          this.viteServer.moduleGraph.invalidateModule(m);
-        }
-      } else {
-        this.viteServer.moduleGraph.invalidateAll();
-      }
+      await this.viteServer.moduleGraph.invalidateAll();
```

Comment:
Last change: when something threw an error, we were doing too much work to only invalidate some modules. If something has erred, just invalidate everything 🤷🏻.